pkg/grpc/podstore/podstore.go

```diff
@@ -246,7 +246,7 @@ func (s store) MarkPodFailed(_ context.Context, req *podstore_protos.MarkPodFail
 	// we don't really need the CAS properties of MutateStatus but there
 	// should be only one system trying to write the status record so it
 	// doesn't hurt.
-	err = s.podStatusStore.MutateStatus(podUniqueKey, mutator)
+	err = s.podStatusStore.MutateStatus(ctx, podUniqueKey, mutator)
```

Comment:
How will this transaction be committed? In `MarkPodFailed` or its callers (in which case we need to play "find the `Trxer` game again)?